libraries/eosiolib/contracts/eosio/permission.hpp

```diff
@@ -136,7 +137,7 @@ namespace eosio {
                                    name                               permission,
                                    const std::set<public_key>&        provided_keys,
                                    const std::set<permission_level>&  provided_permissions = std::set<permission_level>(),
-                                   uint64_t                           provided_delay_us = static_cast<uint64_t>(std::numeric_limits<int64_t>::max())
+                                   microseconds                       provided_delay_us = microseconds{std::numeric_limits<int64_t>::max()}
                                  )
    {
```

Comment:
I think the `provided_delay_us` argument should be renamed to `provided_delay` and the following two lines should be added to the start of this function:
```cpp
int64_t provided_delay_us = provided_delay.count();
eosio::check( provided_delay_us >= 0, "negative delay is not allowed" );
```

The usage of `provided_delay_us` below would also then need to be replaced with `static_cast<uint64_t>(provided_delay_us)`.

Reply:
got it.